core/key.go

```diff
@@ -401,6 +529,27 @@ func (k CollectionSchemaVersionKey) ToDS() ds.Key {
 	return ds.NewKey(k.ToString())
 }
 
+func (k CollectionIndexKey) ToString() string {
+	result := COLLECTION_INDEX
+
+	if k.CollectionName != "" {
+		result = result + "/" + k.CollectionName
+		if k.IndexName != "" {
```

Comment:
todo: It is non-obvious as to why we only want to append `IndexName` if `CollectionName` is not empty - could you please either change this so that `IndexName` is not dependent on `CollectionName`, or document why it is?

Reply:
> But then when asked to decompose from a string it does a lot of assumptions about the elements of the string

It does this because it is safe to do so, and if it doesnt follow the expected format something else is very broken.

> this comment is somewhat related to ...

Whatever gets decided there (I'm happy as-is) should also be documented here. It will likely save developer time in the future.